.github/actions/install-gcc/action.yml

```diff
@@ -0,0 +1,17 @@
+---
+name: "Install GCC"
+description: "Installs GCC on CI runners"
+inputs:
+  version:
+    description: "Which version of GCC to install. E.g. '13' or 'gcc-13'"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Install GCC
+      shell: bash
+      run: |
+        GCC_VERSION=`echo ${{ inputs.version }} | cut -d'-' -f 2`
+        sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
```

Comment:
I updated this somewhere recently to only add the PPA if gcc+14 is not available in the main repos.
Makes running on Ubuntu 24.04+ so much faster.